ocaml/libs/stunnel/stunnel.ml

```diff
@@ -287,6 +296,50 @@ let disconnect ?(wait = true) ?(force = false) x =
    exception instead *)
 exception Stunnel_initialisation_failed
 
+let with_client_config_file ~config f =
+  let tmpfile, tmpch =
+    Filename.open_temp_file ~mode:[Open_text] "stunnel-client-proxy-" ".conf"
+  in
+  Xapi_stdext_pervasives.Pervasiveext.finally
+    (fun () -> output_string tmpch config ; close_out tmpch ; f tmpfile)
+    (fun () -> Unixext.unlink_safe tmpfile)
+
+let with_client_proxy verify_cert r_host r_port l_host l_port f =
```

Comment:
I assume the `l` and `r` prefix should denote `local` and `remote` - I'd prefer longer names for parameters and maybe even named parameters to avoid mixing them up at the call site.

Reply:
> I am not sure that this protects against leaking tmpch. If an exception occurs before close_out, it will leak.
The `finally` can handle the exception case.